pkg/loop/internal/relayer/pluginprovider/ext/ccip/README.md

```diff
@@ -0,0 +1,27 @@
+The code herein is the gRPC wrapper for the CCIP Reporting Plugin `Provider`s
+
+Developing the CCIP Providers requires the follow workflow
+
+1. Update the golang interfaces in `chainlink-common/pkg/types`, and/or  `chainlink-common/pkg/types/ccip`
+2. Mirror those changes as protobuf changes in `chainlink-common/pkg/loop/internal/pb/ccip`
+    note: golang `interface` <-> gRPC `service` and `struct` <-> `message`
+    remember to run `make generate` at the top of the repo to generate your changes
+    see docs in `internal/pb` for more details
+3. If you are adding a new interface, create a file here and implement the gRPC server and client. Follow the testing pattern in the  `test` subdir to ensure roundtrip de/serialization of each and every rpc
+4. If you are updating an interface, then update both the `GRPCClient` and `GRPCServer` in the relevant file.
+    - The client code will not compile if you change the interface in `chainlink-common/pkg/types/ccip` and so these are obvious
+    - However the server code will compile and you will receive runtime `Unimplemented` errors (this is a consequence of golang gRPC best practices that embed a `UnimplementedServer`. the curious reader can go here)
```

Comment:
For debug you can comment out the `UnimplementedServer` to see what methods are missing/broken